src/query/expression/src/date_helper.rs

```diff
@@ -161,3 +162,104 @@ impl AddTimesImpl {
         check_timestamp(ts.wrapping_add(delta.as_() * factor * 1_000_000))
     }
 }
+
+pub trait ToNumber<N> {
+    fn to_number(dt: &DateTime<Tz>) -> N;
+}
+
+pub struct ToNumberImpl;
+
+impl ToNumberImpl {
+    pub fn eval_timestamp<T: ToNumber<R>, R>(ts: i64, tz: &Tz) -> R {
+        let dt = ts.to_timestamp(tz);
+        T::to_number(&dt)
+    }
+
+    pub fn eval_date<T: ToNumber<R>, R>(date: i32, tz: &Tz) -> R {
+        let dt = date.to_date(tz).and_hms(0, 0, 0);
+        T::to_number(&dt)
+    }
+}
+
+pub struct ToYYYYMM;
+pub struct ToYYYYMMDD;
+pub struct ToYYYYMMDDHHMMSS;
+pub struct ToYear;
+pub struct ToMonth;
+pub struct ToDayOfYear;
+pub struct ToDayOfMonth;
+pub struct ToDayOfWeek;
+pub struct ToHour;
+pub struct ToMinute;
+pub struct ToSecond;
+
+impl ToNumber<u32> for ToYYYYMM {
+    fn to_number(dt: &DateTime<Tz>) -> u32 {
+        dt.year() as u32 * 100 + dt.month()
+    }
+}
+
+impl ToNumber<u32> for ToYYYYMMDD {
+    fn to_number(dt: &DateTime<Tz>) -> u32 {
+        dt.year() as u32 * 10000 + dt.month() * 100 + dt.day()
+    }
+}
+
+impl ToNumber<u64> for ToYYYYMMDDHHMMSS {
+    fn to_number(dt: &DateTime<Tz>) -> u64 {
+        dt.year() as u64 * 10000000000
+            + dt.month() as u64 * 100000000
+            + dt.day() as u64 * 1000000
```

Comment:
Use commas to split long numbers.   `1,000,000`

Reply:
maybe `1_000_000`?